procdiscovery/pkg/inspectors/nginx/nginx.go

```diff
@@ -11,14 +14,45 @@ import (
 // but in order to avoid huge refactoring we are adding it as a language for now
 type NginxInspector struct{}
 
-const NginxProcessName = "nginx"
+const (
+	NginxProcessName  = "nginx"
+	NginxVersionRegex = `nginx/(\d+\.\d+\.\d+)`
+)
 
-func (j *NginxInspector) Inspect(p *process.Details) (common.ProgramLanguageDetails, bool) {
-	var programLanguageDetails common.ProgramLanguageDetails
+func (j *NginxInspector) Inspect(p *process.Details) (common.ProgrammingLanguage, bool) {
 	if strings.Contains(p.CmdLine, NginxProcessName) || strings.Contains(p.ExeName, NginxProcessName) {
-		programLanguageDetails.Language = common.NginxProgrammingLanguage
-		return programLanguageDetails, true
+		return common.NginxProgrammingLanguage, true
+	}
+
+	return "", false
+}
+
+func (j *NginxInspector) GetRuntimeVersion(p *process.Details, containerURL string) *version.Version {
+	nginxVersion, err := GetNginxVersion(containerURL)
+	if err != nil {
+		return nil
+	}
+
+	return common.GetVersion(nginxVersion)
+}
+
+func GetNginxVersion(containerURL string) (string, error) {
```

Comment:
The `error` in return types is always `nil`, do we want to return errors from this function when it fails?